knative-operator/pkg/common/eventing.go

```diff
@@ -0,0 +1,25 @@
+package common
+
+import (
+	eventingv1alpha1 "knative.dev/eventing-operator/pkg/apis/eventing/v1alpha1"
+	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func MutateEventing(ke *eventingv1alpha1.KnativeEventing, c client.Client) error {
+	stages := []func(*eventingv1alpha1.KnativeEventing, client.Client) error{
+		eventingImagesFromEnviron,
+	}
+	for _, stage := range stages {
+		if err := stage(ke, c); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// eventingImagesFromEnviron overrides registry images
+func eventingImagesFromEnviron(ke *eventingv1alpha1.KnativeEventing, _ client.Client) error {
+	updateImagesFromEnviron((*servingv1alpha1.Registry)(&ke.Spec.Registry))
```

Comment:
There are 2 `Registry` types, one in eventing one in serving upstream operators.

As we will have a unified type when the operators are merged and since these 2 `Registry` types are exactly the same, I made the `updateImagesFromEnviron` accept the serving `Registry` and I simply cast the eventing one to that.

Reply:
I don't love that solution, tbh. I was thinking the common function would just return a `map[string]string]` that the caller could assign to its registry. The `default` feature complicates that, but there's no harm in returning it in the map, and just let the caller set it if it finds it. Maybe?

Reply:
Thanks for the suggestion. https://github.com/openshift-knative/serverless-operator/pull/126/commits/52ac732151382a5cf78d3d43db7336e13815941c